src/main/java/com/diffmin/Main.java

```diff
@@ -0,0 +1,39 @@
+package com.diffmin;
+
+import com.diffmin.util.Pair;
+import gumtree.spoon.diff.Diff;
+import java.io.File;
+import java.io.FileNotFoundException;
+import spoon.reflect.CtModel;
+
+/** Main execution of generating and applying patch */
+class Main {
+    private final App app;
+
+    private final CtModel patchedCtModel;
+
+    /**
+     * Runs the patch function and generates a patched model.
+     *
+     * @param prevFile Previous version of the file
+     * @param newFile Modified version of the file
+     * @throws FileNotFoundException Exception is raised when path of the file is invalid
+     */
+    Main(File prevFile, File newFile) throws FileNotFoundException {
+        app = new App();
+        Pair<Diff, CtModel> diffAndModel = App.computeDiff(prevFile, newFile);
+        app.generatePatch(diffAndModel.getFirst());
+        app.applyPatch();
+        patchedCtModel = diffAndModel.getSecond();
+    }
+
+    /** Returns the patched model. */
+    public CtModel getModel() {
+        return patchedCtModel;
+    }
+
+    /** Pretty-prints the patched model. */
+    public String displayModel() {
+        return app.displayModifiedModel(patchedCtModel);
+    }
+}
```

Comment:
I have three concerns with this design.

1. It's unexpected that the `main` method is in `App`, and not in `Main`.
2. It's unexpected that a constructor does heavy computation. In general, constructors should setup initial state, not run the entire program.
    - In fact, I wouldn't expect a `Main` class to have a constructor at all, just a `main` method.
3. There is crosslinking between `App` and `Main`; they both refer to each other. There's really no need for them to do so.

I suggest you change it like so:

1. Move the `main` method from `App` to `Main`.
2. Turn the `Main` constructor into a static method, which is called from the `main` method.
    - Have that method return whatever is needed in the `main` method.
3. Don't reference `Main` from `App`.